src/main/java/erb/RouteImpl.java

```diff
@@ -95,23 +145,24 @@ public int getSortTag() {
 	 */  
 	@Override
 	public int compareTo(Route other) {
-			if(sortTag == Driver.SORTBY_ID) {
-				// sort using long 
-				return (Long.valueOf(routeId)).compareTo((Long.valueOf(other.getRouteId())));
+			if(sortTag == _SORTBY_ROUTE_ID) {
+                return (int) (routeId - other.getRouteId());
 			}
 			else if(sortTag == _SORTBY_ROUTE_NAME) {
 				return routeName.compareTo(other.getRouteName());
 			}
 			else if(sortTag == _SORTBY_LENGTH) {
 			     // sort using double
-	             return (Double.valueOf(length)).compareTo((Double.valueOf(other.getLength())));
+                return (int) (length - other.getLength());
```

Comment:
the compare requires multiple clicks in UI since they integer values come back the same, I think the Double class handles this for us
```suggestion
                return Double(length).compareTo((Double)other.getLength());
```

Reply:
Fixed this, just tested and another whoops not sorting doubles correctly.  			    return (Double.valueOf(length)).compareTo((Double.valueOf(other.getLength())));